ecr.tf

```diff
@@ -0,0 +1,61 @@
+# Create secret in AWS SecretsManager to store credentials for upstream repo
+resource "aws_secretsmanager_secret" "ecr_pullthroughcache_dspacecloudreleases" {
+  count                   = var.ecr_pullthrough_cache_rule_config.enable && !var.ecr_pullthrough_cache_rule_config.exist ? 1 : 0
+  name                    = "ecr-pullthroughcache/dspacecloudreleases"
+  recovery_window_in_days = 7
+  tags                    = var.tags
+}
+
+# Store data inside created secret
+resource "aws_secretsmanager_secret_version" "ecr_credentials" {
+  count     = var.ecr_pullthrough_cache_rule_config.enable && !var.ecr_pullthrough_cache_rule_config.exist ? 1 : 0
+  secret_id = aws_secretsmanager_secret.ecr_pullthroughcache_dspacecloudreleases[0].id
+  secret_string = jsonencode(
+    {
+      username    = null
+      accessToken = null
+    }
+  )
+}
+
+# Create pull-through rule for private ECR registry
+resource "aws_ecr_pull_through_cache_rule" "dspacecloudreleases" {
+  count                 = var.ecr_pullthrough_cache_rule_config.enable && !var.ecr_pullthrough_cache_rule_config.exist ? 1 : 0
+  ecr_repository_prefix = "dspacecloudreleases"
+  upstream_registry_url = "dspacecloudreleases.azurecr.io"
+  credential_arn        = aws_secretsmanager_secret.ecr_pullthroughcache_dspacecloudreleases[0].arn
+}
+
+# Data to create custom IAM policy (give permissions to create repositories and replicate images)
+data "aws_iam_policy_document" "eks_node_custom_inline_policy" {
+  count = var.ecr_pullthrough_cache_rule_config.enable ? 1 : 0
+  statement {
+    actions = [
+      "ecr:CreateRepository",
+      "ecr:ReplicateImage",
+      "ecr:BatchImportUpstreamImage"
+    ]
+
+    resources = ["*"]
+  }
+}
+
+# Create IAM policy
+resource "aws_iam_policy" "ecr_policy" {
+  count       = var.ecr_pullthrough_cache_rule_config.enable ? 1 : 0
+  name        = "ecr_pullthrough_policy"
```

Comment:
Could the same name cause a problem if we will deploy the reference infrastructure twice to the same AWS account?

Reply:
Yes it would cause a problem, therefore, random suffix is added to the policy name to prevent this.